contracts/UmaConditionalTokensBinaryAdapter.sol

```diff
@@ -156,162 +156,136 @@ contract UmaCtfAdapter is Auth, BulletinBoard, ReentrancyGuard {
-        if (!isQuestionInitialized(questionID)) {
+    /// @notice Checks whether a questionID is ready to be resolved
+    /// @param questionID - The unique questionID
+    function readyToResolve(bytes32 questionID) public view returns (bool) {
+        if (!isInitialized(questionID)) {
             return false;
         }
 
         QuestionData storage questionData = questions[questionID];
 
-        // Ensure question has not been resolved
-        if (questionData.resolved) {
-            return false;
-        }
-
-        // Ensure question has not been settled
-        if (questionData.settled != 0) {
-            return false;
-        }
-
-        // If the question is disputed by the DVM, do not wait for DVM resolution
-        // instead, immediately flag the question as ready to be settled
-        if (_isDisputed(questionData)) {
-            return true;
-        }
-
         // Check that the OO has an available price
-        return
-            optimisticOracle.hasPrice(
-                address(this),
-                UmaConstants.YesOrNoIdentifier,
-                questionData.requestTimestamp,
-                questionData.ancillaryData
-            );
+        return _hasPrice(questionData);
     }
 
-    /// @notice Settle the question
-    /// Settling a question means that:
-    /// 1. There is an undisputed price available from the OO and so the question can move on to resolution
-    /// 2. The question has been disputed, and a new price request needs to be sent out for the question
+    /// @notice Resolves a question
+    /// Pulls price information from the OO and resolves the underlying CTF market.
+    /// Is only available after price information is available on the OO
     /// @param questionID - The unique questionID of the question
-    function settle(bytes32 questionID) external nonReentrant {
-        require(readyToSettle(questionID), AdapterErrors.NotReadyToSettle);
+    function resolve(bytes32 questionID) external {
+        require(readyToResolve(questionID), AdapterErrors.NotReadyToResolve);
 
         QuestionData storage questionData = questions[questionID];
-        require(!questionData.paused, AdapterErrors.Paused);
 
-        // If the question is disputed, reset the question
-        if (_isDisputed(questionData)) {
-            return _reset(questionID, questionData);
-        }
+        require(!questionData.paused, AdapterErrors.Paused);
+        require(!questionData.resolved, AdapterErrors.AlreadyResolved);
 
-        return _settle(questionID, questionData);
+        // Resolve the underlying market
+        return _resolve(questionID, questionData);
     }
 
-    /// @notice Retrieves the expected payout of a settled question
+    /// @notice Retrieves the expected payout array of the question
     /// @param questionID - The unique questionID of the question
     function getExpectedPayouts(bytes32 questionID) public view returns (uint256[] memory) {
```

Comment:
is `getExpectedPayouts` even needed any longer?

Reply:
This is a fair question, I'm not really sure. It's a nice utility function but it's not necessary anymore